content/en/docs/tasks/job/intra-job-pod-networking-using-pod-hostnames.md

```diff
@@ -0,0 +1,108 @@
+---
+title: Intra-Job Pod Networking Using Pod Hostnames
+content_type: task
+min-kubernetes-server-version: v1.21
+weight: 30
+---
+
+<!-- overview -->
+
+In this example, we will run a job in [indexed completion mode](https://kubernetes.io/blog/2021/04/19/introducing-indexed-jobs/) configured such that
+the pods created by the job can communicate with each other using pod hostnames rather than pod IPs.
+
+Pods within a Job might need to communicate among themselves. They could query the Kubernetes API
+to learn the IPs of the other Pods, but it's much simpler to rely on Kubernetes' built-in DNS resolution.
+Jobs in indexed completion mode automatically set the pod hostname to be in the format of
+`${jobName}-${completionIndex}`, which can be used to deterministically determine
+pod hostnames and enable pod networking *without* needing to create a client connection to
+the Kubernetes control plane to obtain pod hostnames/IPs via API requests. This can be useful
+for use cases where pod networking is required but we don't want to depend on a network 
+connection with the Kubernetes API server.
+
+## {{% heading "prerequisites" %}}
+
+You should already be familiar with the basic use of [Job](/docs/concepts/workloads/controllers/job/).
+
+{{< include "task-tutorial-prereqs.md" >}} {{< version-check >}}
+
+
+<!-- steps -->
+
+## Starting a job with pod networking using pod hostnames
+
+To enable pod-to-pod communication using pod hostnames, you must do the following:
+
+1. Set up a [headless service](https://kubernetes.io/docs/concepts/services-networking/service/#headless-services)
+with a valid label selector for the pods created by your job. This will trigger the 
+Kubernetes' service discovery mechanism to cache the hostnames of 
+the pods running your job (note that the service discovery mechanism does not need to be 
```

Comment:
what do you mean here? The Service or things like kube-dns?

The Service does need to be in the same namespace.

Reply:
I meant the DNS server (coredns etc.), updated doc to clarify this.